api/src/services/platform-service.ts

```diff
@@ -259,4 +307,180 @@ export class PlatformService extends DBService {
+      });
+      const { artifact_id } = await this._submitArtifactToBioHub(artifact);
+
+      return this.publishService.insertProjectAttachmentPublishRecord({
+        artifact_id,
+        project_attachment_id: attachment.id
+      });
+    });
+
+    const reportArtifactPromises = reportAttachments.map(async (attachment) => {
+      const artifact = await this._makeArtifactFromAttachment({ dataPackageId, attachment, file_type: 'Report' });
+      const { artifact_id } = await this._submitArtifactToBioHub(artifact);
+
+      return this.publishService.insertProjectReportPublishRecord({
+        artifact_id,
+        project_report_attachment_id: attachment.id
+      });
+    });
+
+    return [...(await Promise.all(attachmentArtifactPromises)), ...(await Promise.all(reportArtifactPromises))];
+  }
+
+  /**
+   * Uploads the given survey attachments and report attachments to BioHub.
+   *
+   * @param {string} dataPackageId The dataPackageId for the artifact submission
+   * @param {number} surveyId The ID of the survey
+   * @param {number[]} attachmentIds The particular IDs of the attachments to submit to BioHub
+   * @param {number[]} reportAttachmentIds The particular IDs of the report attachments to submit to BioHub
+   * @returns {*} {Promise<{artifact_id: number}[]>} The IDs of all the artifact records in BioHub
+   *
+   * @memberof PlatformService
+   */
+  async uploadSurveyAttachmentsToBioHub(
+    dataPackageId: string,
+    surveyId: number,
+    attachmentIds: number[],
+    reportAttachmentIds: number[]
+  ): Promise<({ survey_attachment_publish_id: number } | { survey_report_publish_id: number })[]> {
+    const attachments = await this.attachmentService.getSurveyAttachmentsByIds(surveyId, attachmentIds);
+    const reportAttachments = await this.attachmentService.getSurveyReportAttachmentsByIds(
+      surveyId,
+      reportAttachmentIds
+    );
+
+    const attachmentArtifactPromises = attachments.map(async (attachment) => {
+      const artifact = await this._makeArtifactFromAttachment({
+        dataPackageId,
+        attachment,
+        file_type: attachment.file_type || 'Other'
+      });
+      const { artifact_id } = await this._submitArtifactToBioHub(artifact);
+
+      return this.publishService.insertSurveyAttachmentPublishRecord({
+        artifact_id,
+        survey_attachment_id: attachment.id
+      });
+    });
+
+    const reportArtifactPromises = reportAttachments.map(async (attachment) => {
+      const artifact = await this._makeArtifactFromAttachment({ dataPackageId, attachment, file_type: 'Report' });
+      const { artifact_id } = await this._submitArtifactToBioHub(artifact);
+
+      return this.publishService.insertSurveyReportPublishRecord({
+        artifact_id,
+        survey_report_attachment_id: attachment.id
+      });
+    });
+
+    return [...(await Promise.all(attachmentArtifactPromises)), ...(await Promise.all(reportArtifactPromises))];
```

Comment:
Could also write this as:
```
return Promise.all([...attachmentArtifactPromises, ...reportArtifactPromises]);
```
Should yield the same result

Reply:
When I write `Promise.all([...attachmentArtifactPromises, ...reportArtifactPromises])`, the compiler complains... Admitedly it looks a bit messy, so I'll refactor